application/F3DStarter.cxx

```diff
@@ -798,45 +798,80 @@ int F3DStarter::Start(int argc, char** argv)
     this->Internals->ApplyPositionAndResolution();
 
     f3d::interactor& interactor = this->Internals->Engine->getInteractor();
+
+    interactor.addCommandCallback("load_previous_file_group",
+      [this](const std::string&) { return this->LoadRelativeFileGroup(-1); });
+
+    interactor.addCommandCallback("load_next_file_group",
+      [this](const std::string&) { return this->LoadRelativeFileGroup(+1); });
+
+    interactor.addCommandCallback("reload_current_file_group",
+      [this](const std::string&) { return this->LoadRelativeFileGroup(0, true, true); });
+
+    interactor.addCommandCallback("add_current_directories",
+      [this](const std::string&) -> bool
+      {
+        if (this->Internals->LoadedFiles.size() > 0)
+        {
+          for (const auto& parentPath : F3DInternals::ParentPaths(this->Internals->LoadedFiles))
+          {
+            this->AddFile(parentPath, true);
+          }
+          return this->LoadRelativeFileGroup(0);
+        }
+        return true;
+      });
+
+    interactor.addCommandCallback("take_screenshot",
+      [this](const std::string& filename) -> bool
+      {
+        // TODO add test
+        std::string localFilename =
```

Comment:
what if the provided filename is invalid? Is it checked in `SaveScreenshot`?

Reply:
of course

Reply:
looking at the code, `fs::create_directories` is not checked . Should be fixed but unrelated to this PR.